### DISCUSSION TITLE
Enum comparison

### DISCUSSION THREAD

**Discussion by einarwar** [astral-sh/ruff#11617] [Q&A]
Hi

According to the [python documentation](https://docs.python.org/3/howto/enum.html#comparisons) enums are compared by identity, ie. `MyEnum.A is MyEnum.A`. However, i was unable to find any tools or rules related to enum-comparison in ruff.

I was wondering if any existing tools/rules implement this check, or if it exists as an not yet implemented part of some tools?

**Comment by tjkuson** [comment]
I don't think so. Ruff would need to support better type-inference to distinguish between Enums and other objects (#3893).
